DataStructures/BinarySearchTree/BinarySearchTree.cs

```diff
@@ -0,0 +1,275 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace DataStructures.BinarySearchTree
+{
+    /// <summary>
+    /// Node-based binary tree data structure which has the following properties:
+    /// - The left subtree of a node contains only nodes with keys lesser than the node’s key
+    /// - The right subtree of a node contains only nodes with keys greater than the node’s key
+    /// - The left and right subtree each must also be a binary search tree.
+    /// Source: https://www.geeksforgeeks.org/binary-search-tree-set-1-search-and-insertion/.
+    /// </summary>
+     /// <typeparam name="T">Generic type taht have to be comparable.</typeparam>
+    public class BinarySearchTree<T>
+    {
+        /// <summary>
+        /// Current root of a subtree.
+        /// </summary>
+        private Node<T>? rootNode;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BinarySearchTree{T}"/> class.
+        /// </summary>
+        public BinarySearchTree()
```

Comment:
Codacy found an issue: [Remove this redundant constructor.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6278013)